dbusmock/templates/bluez5.py

```diff
@@ -393,3 +498,625 @@ def DisconnectDevice(_self, adapter_device_name, device_address):
+def AddBeacon(self,
+              adapter_device_name='hci0',
+              device_address='11:01:02:03:04:05',
+              manf_id=None,
+              manf_data=None,
+              service_uuid=None,
+              service_data=None,
+              ):
+    """Convenience method to add a Bluetooth device acting as AltBeacon
+
+    You have to specify a device address which must be a valid Bluetooth
+    address (e.g. 'AA:BB:CC:DD:EE:FF'). The alias is the human-readable name
+    for the device (e.g. as set on the device itself), and the adapter device
+    name is the device_name passed to AddAdapter.
+
+    This will create a new, unpaired and unconnected device.
+
+    Returns the new object path.
+    """
+    device_name = 'dev_' + device_address.replace(':', '_').upper()
+    adapter_path = '/org/bluez/' + adapter_device_name
+    path = adapter_path + '/' + device_name
+
+    if adapter_path not in mockobject.objects:
+        raise dbus.exceptions.DBusException(
+            'Adapter %s does not exist.' % adapter_device_name,
+            name=BLUEZ_MOCK_IFACE + '.NoSuchAdapter')
+
+    properties = {
+        'UUIDs': dbus.Array([], signature='s', variant_level=1),
+        'Blocked': dbus.Boolean(False, variant_level=1),
+        'Connected': dbus.Boolean(False, variant_level=1),
+        'LegacyPairing': dbus.Boolean(False, variant_level=1),
+        'Paired': dbus.Boolean(False, variant_level=1),
+        'Trusted': dbus.Boolean(False, variant_level=1),
+        'RSSI': dbus.Int16(-61, variant_level=1),  # arbitrary
+        'Adapter': dbus.ObjectPath(adapter_path, variant_level=1),
+        'Address': dbus.String(device_address, variant_level=1),
+        'AddressType': dbus.String("random"),
+        'Alias': dbus.String("40-A1-82-A6-BB-3D", variant_level=1),
+    }
+    if service_uuid:
+        properties['UUIDs'].append(service_uuid)
+        properties['ServiceData'] = dbus.Dictionary({service_uuid: service_data})
+    if manf_id:
+        properties['ManufacturerData'] = dbus.Dictionary({manf_id: manf_data})
+    self.AddObject(path,
+                   DEVICE_IFACE,
+                   # Properties
+                   properties,
+                   # Methods
+                   [
+                       ('CancelPairing', '', '', ''),
+                       ('Connect', '', '', ""),
+                       ('ConnectProfile', 's', '', ''),
+                       ('Disconnect', '', '', ''),
+                       ('DisconnectProfile', 's', '', ''),
+                       ('Pair', '', '', ''),
+                   ])
+
+    manager = mockobject.objects['/']
+    manager.EmitSignal(OBJECT_MANAGER_IFACE, 'InterfacesAdded',
+                       'oa{sa{sv}}', [
+                           dbus.ObjectPath(path),
+                           {DEVICE_IFACE: properties},
+                       ])
+
+
+@dbus.service.method(BLUEZ_MOCK_IFACE,
+                     in_signature='', out_signature='')
+def DeviceDiscovery(self):
```

Comment:
I don't think that we should be making beacons or devices appear by default when a discovery starts. It might however be a good idea to have a list of devices/beacons that would be discovered, which you could add to using a method on the Mock interface.

Each one of those beacon types you list here could be a separate beacon one could add, something like:
```
def AddDiscoveredDevice(self, beacon_type):
    if beacon_type == 'eddystone-url':
...
```